src/lib/components/dark-mode-navigation-button.svelte

```diff
@@ -0,0 +1,26 @@
+<script>
+  import NavigationButton from '$lib/holocene/navigation/navigation-button.svelte';
+  import { translate } from '$lib/i18n/translate';
+  import {
+    getNextDarkModePreference,
+    useDarkMode,
+    useDarkModePreference,
+  } from '$lib/utilities/dark-mode';
+
+  $: buttonText =
+    $useDarkModePreference == null
+      ? translate('common.system-default')
+      : $useDarkModePreference
+        ? translate('common.night')
+        : translate('common.day');
+</script>
+
+<NavigationButton
+  onClick={() =>
+    ($useDarkModePreference = getNextDarkModePreference(
+      $useDarkModePreference,
+    ))}
```

Comment:
Generally I think we prefer to pull these out into functions

const onClick = () => {
 $useDarkModePreference = getNextDarkModePreference($useDarkModePreference)
}